### ISSUE TITLE
Problems about re-produce the  training results  in  QF-estimation loss

### ISSUE THREAD

**Issue by JuZiSYJ** [jiaxi-jiang/FBCNN#5]
Hi, It is a nice job to enable flexible QF embedding in JPEG deblocking!
I want to re-produce the results following the directions in the paper, However, there are some questions.

1. qf_loss
Following the paper, I calculate the loss as:
````          
            input = batch["degree"].type(Tensor)
            label = batch["label"].type(Tensor)
            qf_label = batch["qf"].type(Tensor)
            qf_label = (1-qf_label/100) # 0-1
            out, qf_pred = model(input)

            mse = mse_pixelwise(out, label)

            cls_loss = l1_pixelwise2(qf_pred, qf_label)

            loss = l1_pixelwise(out, label) * 0.5  + mse * 0.5 + cls_loss * 0.1
````
However, during the  training phase, the cls_loss is not changed at all
````
2021-10-09 22:38:47,481 - __main__ - INFO - 
[QF 10 lr 0.000100 cls_loss:0.23448  loss:0.04852 Epoch 0/120] [Batch 19/203] [psnr: 19.300194 ]  ETA: 6:18:37.362892   [mse: 0.005858 ] [mse_original: 0.000583 ]
2021-10-09 22:39:08,319 - __main__ - INFO - 
[QF 10 lr 0.000100 cls_loss:0.20090  loss:0.03645 Epoch 0/120] [Batch 39/203] [psnr: 21.449274 ]  ETA: 7:02:01.486693   [mse: 0.003679 ] [mse_original: 0.000384 ]
2021-10-09 22:39:29,365 - __main__ - INFO - 

2021-10-10 05:11:26,158 - __main__ - INFO - 
[QF 10 lr 0.000020 cls_loss:0.20545  loss:0.02607 Epoch 119/120] [Batch 159/203] [psnr: 34.804854 ]  ETA: 0:00:41.660994   [mse: 0.000345 ] [mse_original: 0.000475 ]
2021-10-10 05:11:45,303 - __main__ - INFO - 
[QF 10 lr 0.000020 cls_loss:0.22516  loss:0.02851 Epoch 119/120] [Batch 179/203] [psnr: 34.770472 ]  ETA: 0:00:22.570621   [mse: 0.000385 ] [mse_original: 0.000513 ]
2021-10-10 05:12:04,498 - __main__ - INFO - 
[QF 10 lr 0.000020 cls_loss:0.18704  loss:0.02385 Epoch 119/120] [Batch 199/203] [psnr: 34.775089 ]  ETA: 0:00:03.771435   [mse: 0.000276 ] [mse_original: 0.000377 ]
2021-10-10 05:12:12,970 - __main__ - INFO - QF: 10 [PSNR: live1: 29.005289    classic: 29.159697]  
 [SSIM: live1: 0.806543    classic: 0.795164] 
 [max PSNR in live1: 29.005289, max epoch: 119]
````
The cls_loss is always at 2,  and when I test the qf estimation with input image in qf=[10,20,..,90], the qf estimation is always 53, 

which shows that the cls_loss does not work well. It is reasonable that qf estimation is 50 since it is good for L1 loss when not trained well.

Due to the GPU limitation, I set the Batchsize = 96,  and training with DIV2K patches for about 2.5W iters

Is there anything wrong in my implementation, thanks!

**Comment by JuZiSYJ** [comment]
Thanks a lot！

I ignore the slightly different shape (e.g. [64, 1] and [64]) in cls_Loss

After I change the code to 
````
qf_label = (1-qf_label/100).unsqueeze(1)
````

The cls_loss reduced to 0.0x after 20 epochs

````
[QF 10 lr 0.000100 cls_loss:0.03812  loss:0.01708 Epoch 24/400] [Batch 219/305] [psnr: 34.170276 ]  ETA: 11:58:01.264007   [mse: 0.000472 ] [mse_original: 0.000565 ]
2021-10-13 13:43:06,999 - __main__ - INFO - 
[QF 10 lr 0.000100 cls_loss:0.04468  loss:0.01648 Epoch 24/400] [Batch 239/305] [psnr: 34.173710 ]  ETA: 11:51:16.305051   [mse: 0.000443 ] [mse_original: 0.000521 ]
2021-10-13 13:43:14,546 - __main__ - INFO - 
[QF 10 lr 0.000100 cls_loss:0.04180  loss:0.01545 Epoch 24/400] [Batch 259/305] [psnr: 34.166023 ]  ETA: 11:58:02.849512   [mse: 0.000348 ] [mse_original: 0.000413 ]
2021-10-13 13:43:22,137 - __main__ - INFO - 
[QF 10 lr 0.000100 cls_loss:0.06484  loss:0.01739 Epoch 24/400] [Batch 279/305] [psnr: 34.182455 ]  ETA: 12:03:49.188979   [mse: 0.000337 ] [mse_original: 0.000405 ]
2021-10-13 13:43:29,664 - __main__ - INFO - 
[QF 10 lr 0.000100 cls_loss:0.03710  loss:0.01539 Epoch 24/400] [Batch 299/305] [psnr: 34.193803 ]  ETA: 11:51:31.754653   [mse: 0.000372 ] [mse_original: 0.000449 ]
2021-10-13 13:43:36,330 - __main__ - INFO - QF: 10 [PSNR: live1: 28.523461    classic: 28.636578]  
 [SSIM: live1: 0.796034    classic: 0.782925] 
 [max PSNR in live1: 28.523461, max epoch: 24]
````

Thanks a lot!
